BYOC public preview docs for Azure/AWS/GCP (#23182)
cockroachdb/docs: src/current/cockroachcloud/byoc-overview.md

{{site.data.alerts.callout_info}}
The BYOC {{ site.data.products.cloud }} deployment option is currently in [Preview]({% link {{ site.current_cloud_version }}/cockroachdb-feature-availability.md %}).
{{site.data.alerts.end}}

## Shared responsibility model for BYOC

In any CockroachDB {{ site.data.products.cloud }} deployment, responsibility for a successful and healthy deployment is [split between you and Cockroach Labs]({% link cockroachcloud/production-checklist.md %}). In a BYOC deployment, all of the [CockroachDB {{ site.data.products.cloud }} infrastructure]({% link cockroachcloud/plan-your-cluster-advanced.md %}#advanced-cluster-architecture) except the control plane lives in an account under your control, which means that you incur additional responsibilities under the shared model.

The following table describes the split of responsibilities between you and Cockroach Labs in the shared responsibility model for BYOC:

Area        | Cockroach Labs responsibility | Customer responsibility
:----------:|:-----------------------------:|:----------------------:

> Uptime      | Ensure 99.999% cluster uptime | Ensure that clusters remain accessible

Reviewer: Ensure that clusters remain accessible -> accesible to CRL via cross-account IAM method documented above?

Author: Done with a slight edit,
"documented above" -> "as documented in the corresponding deployment guide"

(fun fact, both human readers and AI agents don't work well with cardinal directions about page placement within docs)

Deployments | Automate cluster provisioning and scaling, provide hardware best practices | Provision new cloud service accounts and grant IAM permissions for Cockroach Labs to create and manage clusters
Upgrades    | Provide automatic minor/patch upgrades and major upgrade automation via Terraform, APIs, or the {{ site.data.products.cloud }} Console | Initiate [major version upgrades]({% link cockroachcloud/upgrade-cockroach-version.md %}), [set maintenance windows]({% link cockroachcloud/advanced-cluster-management.md %}#set-a-maintenance-window) if applicable
Workload    | Troubleshoot problems as they pertain to cluster availability | [Size clusters]({% link cockroachcloud/advanced-cluster-management.md %}#scale-your-cluster) to manage workload requirements, [tune performance]({% link {{ site.versions["stable"] }}/performance-recipes.md %}), and adjust schema designs with support from Cockroach Labs
Backups     | Initialize a default backup schedule and write to customer-owned Cloud storage, ensure backup jobs run successfully | Configure a backup schedule as needed to meet RPO/RTO requirements

> Support     | Reactively and proactively identify and resolve availability-impacting incidents | Ensure sufficient hardware is made available and appropriate IAM permissions are maintained at all times

Reviewer: Ensure sufficient hardware is made available and appropriate IAM permissions are maintained at all times -> including negotiate cloud resource quotas and availability with CSP

---
title: CockroachDB BYOC Overview
summary: Learn about the Bring-Your-Own-Cloud deployment option for CockroachDB
toc: true
keywords: deployment, byoc
---

> CockroachDB {{ site.data.products.cloud }} supports a ["bring your own cloud" (BYOC) deployment model](https://www.cockroachlabs.com/product/cloud/bring-your-own-cloud/), where CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} is hosted in your own account rather than in an account managed by Cockroach Labs. This model allows you to take more control of security and take advantage of existing cloud service credits or discounts.

Reviewer 2: ```suggestion
CockroachDB {{ site.data.products.cloud }} supports a [Bring Your Own Cloud (BYOC) deployment model](https://www.cockroachlabs.com/product/cloud/bring-your-own-cloud/), where CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} is hosted in your own cloud account rather than in an account managed by Cockroach Labs. This model allows you to take more control of security and take advantage of existing cloud service credits or discounts.
```

Could optionally italicize Bring Your Own Cloud but I probably wouldn't.

{{site.data.alerts.callout_info}}

> The BYOC {{ site.data.products.cloud }} deployment option is currently in [Preview]({% link {{ site.current_cloud_version }}/cockroachdb-feature-availability.md %}).

Reviewer 2: ```suggestion
{% include feature-phases/preview.md %}
```

{{site.data.alerts.end}}

## Shared responsibility model for BYOC

> In any CockroachDB {{ site.data.products.cloud }} deployment, responsibility for a successful and healthy deployment is [split between you and Cockroach Labs]({% link cockroachcloud/production-checklist.md %}). In a BYOC deployment, all of the [CockroachDB {{ site.data.products.cloud }} infrastructure]({% link cockroachcloud/plan-your-cluster-advanced.md %}#advanced-cluster-architecture) except the control plane lives in an account under your control, which means that you incur additional responsibilities under the shared model.

Reviewer 2: Re: "In a BYOC deployment, all of the CockroachDB Cloud infrastructure except the control plane lives in an account under your control."

That sort of feels like it could deserve its own header above this one. Basically an architectural overview that possibly duplicates the diagram that you're linking to here? A "BYOC Overview" feels like it would merit an explanation like this, whereas it feels like a bit of an aside here.

Author: That's good feedback, my 2c is that I want to wait and see what questions customers have about BYOC architecture in practice when they're not being hand-held by an account team like they have so far. There are a few aspects of BYOC that I think could be clarified with diagrams (especially the Azure entity relationships) but want to prioritize that based on actual customer need.

The following table describes the split of responsibilities between you and Cockroach Labs in the shared responsibility model for BYOC:

Area        | Cockroach Labs responsibility | Customer responsibility
:----------:|:-----------------------------:|:----------------------:

> Uptime      | Ensure 99.999% cluster uptime | Ensure that clusters remain accessible to CRL via cross-account IAM as documented in the corresponding deployment guide

Reviewer 2: ```suggestion
Uptime      | Ensure 99.999% cluster uptime | Ensure that clusters remain accessible to CRL via cross-account IAM as documented in the corresponding [deployment guide](#next-steps)
```

Deployments | Automate cluster provisioning and scaling, provide hardware best practices | Provision new cloud service accounts and grant IAM permissions for Cockroach Labs to create and manage clusters
Upgrades    | Provide automatic minor/patch upgrades and major upgrade automation via Terraform, APIs, or the {{ site.data.products.cloud }} Console | Initiate [major version upgrades]({% link cockroachcloud/upgrade-cockroach-version.md %}), [set maintenance windows]({% link cockroachcloud/advanced-cluster-management.md %}#set-a-maintenance-window) if applicable
Workload    | Troubleshoot problems as they pertain to cluster availability | [Size clusters]({% link cockroachcloud/advanced-cluster-management.md %}#scale-your-cluster) to manage workload requirements, [tune performance]({% link {{ site.versions["stable"] }}/performance-recipes.md %}), and adjust schema designs with support from Cockroach Labs
Backups     | Initialize a default backup schedule and write to customer-owned Cloud storage, ensure backup jobs run successfully | Configure a backup schedule as needed to meet RPO/RTO requirements

> Support     | Reactively and proactively identify and resolve availability-impacting incidents | Ensure sufficient hardware is made available, including negotiating cloud resource quotas and availability with your cloud service provider. Maintain appropriate IAM permissions at all times

Reviewer 2: Is the "Customer responsibility" cell of the "Support" column doesn't really feel so relevant to the concept of support.

Author: Clarified that the three points in this column "make hardware available, negotiate resource quotas, maintain IAM permissions" are all in service of reducing support needs for this cluster (don't cause us headaches by not proactively scaling the cluster)

Billing     | Meter vCPUs consumed, [charge for vCPU consumption]({% link cockroachcloud/costs.md %}) at the per-minute level | Negotiate with cloud service provider, manage infrastructure spend and discounts

## Next steps

> CockroachDB supports BYOC deployments in Amazon Web Services, Microsoft Azure, and Google Cloud Platform. Read the corresponding deployment guides:

Reviewer 2: ```suggestion
CockroachDB supports BYOC deployments in Amazon Web Services, Microsoft Azure, and Google Cloud Platform. To prepare your cloud account for a BYOC deployment, refer to the corresponding deployment guide:
```
Rich has sold me on the format "To [do this thing you want to do], [follow this instruction]." for imperative statements like these.

> - [Prepare a CockroachDB Cloud BYOC Deployment in Amazon Web Services]({% link cockroachcloud/byoc-aws-deployment.md %}).

Reviewer 2: I'd get rid of these periods.

- [Prepare a CockroachDB Cloud BYOC Deployment in Azure]({% link cockroachcloud/byoc-azure-deployment.md %}).
- [Prepare a CockroachDB Cloud BYOC Deployment in Google Cloud Platform]({% link cockroachcloud/byoc-gcp-deployment.md %}).

> Once your cloud account is prepared for a BYOC deployment, cluster configuration and management is identical to a CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} cluster. Read more about CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} cluster planning and management:

Reviewer 2: ```suggestion
Once your cloud account is prepared for a BYOC deployment, cluster configuration and management is identical to a CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} cluster. To learn more about CockroachDB {{ site.data.products.cloud }} {{ site.data.products.advanced }} cluster planning and management, refer to the following guides:
```
